
From: Ingo Molnar <mingo@elte.hu>

one final bit missing is a third place in mm/filemap.c that needs 
nosleepcheck:

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/mm/filemap.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN mm/filemap.c~add-a-few-might_sleep-checks-fix mm/filemap.c
--- 25/mm/filemap.c~add-a-few-might_sleep-checks-fix	2004-07-28 22:10:33.337720112 -0700
+++ 25-akpm/mm/filemap.c	2004-07-28 22:10:33.342719352 -0700
@@ -1654,7 +1654,7 @@ __filemap_copy_from_user_iovec(char *vad
 		int copy = min(bytes, iov->iov_len - base);
 
 		base = 0;
-		left = __copy_from_user(vaddr, buf, copy);
+		left = __copy_from_user_inatomic(vaddr, buf, copy);
 		copied += copy;
 		bytes -= copy;
 		vaddr += copy;
_
